utils/policy_roles_adapter.js

```diff
@@ -67,25 +67,32 @@ const MAPPING = {
     restrictPostDelete: {
         all: [
             {roleName: 'channel_user', permission: Permissions.DELETE_POST, shouldHave: true},
-            {roleName: 'channel_admin', permission: Permissions.DELETE_POST, shouldHave: false},
-            {roleName: 'channel_admin', permission: Permissions.DELETE_OTHERS_POSTS, shouldHave: false},
             {roleName: 'team_admin', permission: Permissions.DELETE_POST, shouldHave: true},
             {roleName: 'team_admin', permission: Permissions.DELETE_OTHERS_POSTS, shouldHave: true}
         ],
         team_admin: [
             {roleName: 'channel_user', permission: Permissions.DELETE_POST, shouldHave: false},
-            {roleName: 'channel_admin', permission: Permissions.DELETE_POST, shouldHave: false},
-            {roleName: 'channel_admin', permission: Permissions.DELETE_OTHERS_POSTS, shouldHave: false},
             {roleName: 'team_admin', permission: Permissions.DELETE_POST, shouldHave: true},
             {roleName: 'team_admin', permission: Permissions.DELETE_OTHERS_POSTS, shouldHave: true}
         ],
         system_admin: [
             {roleName: 'channel_user', permission: Permissions.DELETE_POST, shouldHave: false},
-            {roleName: 'channel_admin', permission: Permissions.DELETE_POST, shouldHave: false},
-            {roleName: 'channel_admin', permission: Permissions.DELETE_OTHERS_POSTS, shouldHave: false},
```

Comment:
I removed these mappings because they're the same for all three values so should be default.